### DISCUSSION TITLE
Is there a way to validate or test the results generated by the model using a specific backend?

### DISCUSSION THREAD

**Discussion by wniu9** [ggml-org/llama.cpp#6345] [General]
I am trying to enable `ssm_scan` on unsupported GPU Vulkan backends. While `ssm_scan` can run on the CPU, it is not supported on Vulkan. However, I am unsure how to validate the results of my implementation for `ssm_scan`. Is there a way of cross validating the operators' results (or even intermediate results for a whole graph), e.g., comparing GPU's results with the results generated by CPU backend?

**Comment by ggerganov** [comment]
You need to implement a test(s) in `test-backend-ops` (follow the existing patterns) and make sure the following command succeeds:

```
./test-backend-ops -o SSM_SCAN
```

**Reply by wniu9** [reply]
Thanks! Will definitely have a try.

**Reply by wniu9** [reply]
Btw, If we support a new operator for CPU backend, are we only allowed to verify it by providing ground truth files?

**Reply by ggerganov** [reply]
You can verify it anyway you want. `test-backend-ops` guarantees that the implementation across different backends produces the same result
